test/unit/mem/buf/src/BufTest.cpp

```diff
@@ -264,3 +264,34 @@ TEMPLATE_LIST_TEST_CASE("memBufAccessorAdaptorTest", "[memBuf]", alpaka::test::T
 
     testBufferAccessorAdaptor<Acc>(extent, index);
 }
+
+TEMPLATE_LIST_TEST_CASE("memBufMove", "[memBuf]", alpaka::test::TestAccs)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Pltf = alpaka::Pltf<alpaka::Dev<Acc>>;
+    using Elem = std::size_t;
+
+    auto const dev = alpaka::getDevByIdx<Pltf>(0u);
+    auto const extent = alpaka::Vec<alpaka::DimInt<0>, Idx>{};
+
+    // move constructor
+    {
+        auto buf1 = alpaka::allocBuf<Elem, Idx>(dev, extent);
+        *buf1 = 1;
+        auto buf2{std::move(buf1)};
```

Comment:
at this point `buf1` is invalid, right ?

Reply:
yes. It's a moved from object. But it should destruct fine.